### PR TITLE
Feature/improve jobs

### PR DESCRIPTION
**Pull request by raftx24**
**We cannot customize the queue for finalize & rejected exports!**

### CODE REVIEW

File: src/Services/Importers/Finalize.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        $this->user = $user;
+        $this->params = $params;
+        $this->template = $template;
+    }
+
+    public function handle(): void
+    {
+        $this->afterHook();
+
+        $this->dataImport->refresh()
```
Comment (aocneanu) [inline src/Services/Importers/Finalize.php]: let's move the refresh in the Import class, when we instantiate Finalize

File: src/Services/Importers/Finalize.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
+<?php
```
Comment (aocneanu) [inline src/Services/Importers/Finalize.php:1]: wrap this service in a job, and dispatch the job in batch->then()

The job will use `SerializesModels` so the dataImport->refresh is not longer needed

File: src/Services/Exporters/Rejected.php [hunk 3]
```diff
@@ -30,7 +30,7 @@ public function __construct(DataImport $dataImport, User $user)
         $this->hashFilename = $this->hashFilename();
```
Comment (aocneanu) [inline src/Services/Exporters/Rejected.php:30]: wrap this in a job and run the job in batch->then(). This way the splitting job will not run the export rejected task, and the pipeline won't stay blocked.

**Review by aocneanu — CHANGES_REQUESTED** [review]
With job batching L8 also adds the ability to programatically cancel a batch.

This means that we could use something like
```php
        if ($this->dataImport->status === Statuses::Canceled) {
            $this->batch()->cancel();
        } else {
            (new Chunk(
                $this->dataImport,
                $this->template,
                $this->user,
                $this->params,
                $this->sheetName,
                $this->chunk,
                $this->index
            ))->run();
        }
    }
```

in our ChukImport job

Also we might not need or we may use the same technique in the other places where we were checking for the canceled status

I guess naming the batch something like "importing filename.xlsx" would be good also for horizon

**Comment by raftx24** [comment]
> With job batching L8 also adds the ability to programatically cancel a batch.
> 
> This means that we could use something like
> 
> ```
>         if ($this->dataImport->status === Statuses::Canceled) {
>             $this->batch()->cancel();
>         } else {
>             (new Chunk(
>                 $this->dataImport,
>                 $this->template,
>                 $this->user,
>                 $this->params,
>                 $this->sheetName,
>                 $this->chunk,
>                 $this->index
>             ))->run();
>         }
>     }
> ```
> 
> in our ChukImport job
> 
> Also we might not need or we may use the same technique in the other places where we were checking for the canceled status
> 
> I guess naming the batch something like "importing filename.xlsx" would be good also for horizon

@aocneanu , I guess we cannot remove other places!  
image  
this happens before ->batch->cancel()  
image    
this happens before ->batch->cancel()

**Pull request merged**